src/client/provider.ts

```diff
@@ -0,0 +1,65 @@
+/**
+ * @module Provider
+ */
+
+/**
+ * Need a comment block here so that the comment block above is interpreted as a file comment, and not a comment on the
+ * import below.
+ *
+ * @hidden
+ */
+import semverCompare from 'semver-compare';
+
+import {withStrictTimeout} from '../provider/common/async';
```

Comment:
Client must not import from `src/provider`

Reply:
✅ Moved to `client/async.ts` and made hidden